### PR TITLE
Support for custom texmf folder

### PR DESCRIPTION
**Pull request by hfjn**
Added support for custom texmf folder as discussed in #2

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -30,16 +34,26 @@ printUsage() {
   echo "  -b, --bibtex          Run bibtex"
   echo "  -g, --glossaries      Run makeglossaries"
   echo "  -w, --watch           Watch for changes"
-  echo "      --ignore-errors   Continue watching even if an error occurred"
+  echo "      --ignore-errors   Continue watching even if an error occurred"  
+  echo "  -t, --texmf [PATH]    Custom texmf folder path"
+  echo "  -i, --init [FILE]     Name of the font to be initialized"
```
Comment (mtneug) [inline build.sh:39]: `--map` would probably be a better name (see below)?

File: build.sh [hunk 2]
```diff
@@ -30,16 +34,26 @@ printUsage() {
...
     -file-line-error \
     -synctex 1 \
-    "${FILE}.tex"
+    "${FILE}.tex"; \
+  else \
+    pdflatex \
+      -interaction nonstopmode \
+      -file-line-error \
+      -synctex 1 \
+      "${FILE}.tex"; fi;
```
Comment (mtneug) [inline build.sh:56]: Revert the change and export `TEXMFHOME` once in `main`.

File: build.sh [hunk 3]
```diff
@@ -49,12 +63,23 @@ runMakeglossaries() {
 }
 
 runBibtex() {
-  bibtex -terse "${FILE}.aux"
+  if [[ "${TEXMF}" == "true" ]]; \
+   then TEXMFHOME="${TEXMF_PATH}" bibtex -terse "${FILE}.aux"; \
+   else bibtex -terse "${FILE}.aux"; fi;
```
Comment (mtneug) [inline build.sh:68]: Revert the change and export `TEXMFHOME` once in `main`.

File: build.sh [hunk 4]
```diff
@@ -49,12 +63,23 @@ runMakeglossaries() {
...
+}
+
+initTexmf(){
+  if [[ "${TEXMF}" == "true" ]]; \
+  then TEXMFHOME="${TEXMF_PATH}" mktexlsr \
+  && TEXMFHOME="${TEXMF_PATH}" updmap --enable Map "${INIT_FILE}";\
+  else mtexlsr \
+  && updmap --enable Map "${INIT_FILE}"; fi;\
```
Comment (mtneug) [inline build.sh:76]: No need to connect the commands with `&&` when run once in `main` (see below).

After reading up on `mktexlsr` it seams that it should be run after making any changes to the `texmf` tree. So it should be run if `--texmf` is used:

```sh
enableMap() {
  updmap --enable Map "${MAP_FILE}"
}

initTexmf() {
  mktexlsr
}
```

Both should be called once from `main` if needed.

File: build.sh [hunk 5]
```diff
@@ -49,12 +63,23 @@ runMakeglossaries() {
...
 }
 
 run() {
   {
-       runPdflatex \
+    if [[ "${INIT}" == "true" ]]; then initTexmf; fi \
+    && runPdflatex \
```
Comment (mtneug) [inline build.sh:82]: Revert change and move `initTexmf` call to `main`. Adding it here will execute `initTexmf` on every run when using `watch`.

File: build.sh [hunk 6]
```diff
@@ -30,16 +34,26 @@ printUsage() {
   echo "  -b, --bibtex          Run bibtex"
   echo "  -g, --glossaries      Run makeglossaries"
   echo "  -w, --watch           Watch for changes"
-  echo "      --ignore-errors   Continue watching even if an error occurred"
```
Comment (mtneug) [inline build.sh:33]: Please remove trailing whitespace.

File: build.sh [hunk 7]
```diff
@@ -49,12 +63,23 @@ runMakeglossaries() {
 }
 
 runBibtex() {
-  bibtex -terse "${FILE}.aux"
+  if [[ "${TEXMF}" == "true" ]]; \
+   then TEXMFHOME="${TEXMF_PATH}" bibtex -terse "${FILE}.aux"; \
+   else bibtex -terse "${FILE}.aux"; fi;
+}
+
+initTexmf(){
```
Comment (mtneug) [inline build.sh:71]: Please add a space between `()` and `{`.

File: build.sh [hunk 8]
```diff
@@ -49,12 +63,23 @@ runMakeglossaries() {
...
+  if [[ "${TEXMF}" == "true" ]]; \
+  then TEXMFHOME="${TEXMF_PATH}" mktexlsr \
+  && TEXMFHOME="${TEXMF_PATH}" updmap --enable Map "${INIT_FILE}";\
+  else mtexlsr \
```
Comment (mtneug) [inline build.sh:75]: `mtexlsr` -> `mktexlsr` :wink:

**Review by mtneug — CHANGES_REQUESTED** [review]
Maybe we should add/modify a test document with a texmf tree and a font?

**Comment by hfjn** [comment]
Now working on this in [texmf](https://github.com/mtneug/texlive-docker/tree/texmf). Check state in #2.

**Pull request closed without merging**